reports/performance/_shap.qmd

```diff
@@ -82,49 +82,83 @@ create_shapviz <- function(shap_df, assessment_df, idx) {
     "meta_triad"
   )]
   shapviz::shapviz(
-    object = shap_df %>%
+    object = shap_df_filtered %>%
       select(all_of(shap_predictors)) %>%
       slice(idx) %>%
       as.matrix(),
     X = assessment_df %>%
       select(all_of(shap_predictors)) %>%
       slice(idx),
-    baseline = shap_df$pred_card_shap_baseline_fmv[1]
+    baseline = shap_df_filtered$pred_card_shap_baseline_fmv[1]
   )
 }
 ```
 
-## Riverside Beeswarm
+## Beeswarm Plot
 
-```{r _shap_riverside_beeswarm, eval=shap_exists}
-shap_riverside_idx <- which(
-  assessment_card_filtered$meta_township_code == ccao::town_convert("Riverside")
-)
-create_shapviz(shap_df_filtered, assessment_card_filtered, shap_riverside_idx) %>%
-  shapviz::sv_importance("beeswarm") +
-  theme_minimal()
-```
+```{r}
+shap_predictors <- names(shap_df_filtered)
+shap_predictors <- shap_predictors[!shap_predictors %in% c(
+  "meta_year",
+  "meta_pin",
+  "meta_card_num",
+  "pred_card_shap_baseline_fmv",
+  "township_code",
+  "meta_triad"
+)]
 
-cat("## Calumet Beeswarm")
+townships_to_iterate <- shap_df_filtered %>%
+  distinct(township_code) %>%
+  pull() %>%
+  as.character()
 
-```{r _shap_calumet_beeswarm}
-shap_calumet_idx <- which(
-  assessment_card_filtered$meta_township_code == ccao::town_convert("Calumet")
+shap_idx_full_model <- which(
+  assessment_card_filtered$meta_township_code %in% townships_to_iterate
 )
-create_shapviz(shap_df_filtered, assessment_card_filtered, shap_calumet_idx) %>%
-  shapviz::sv_importance("beeswarm") +
+
+shapviz::shapviz(
+  object = shap_df_filtered %>%
+    select(all_of(shap_predictors)) %>%
+    slice(shap_idx_full_model) %>%
+    as.matrix(),
+  X = assessment_card_filtered %>%
+    select(all_of(shap_predictors)) %>%
+    slice(shap_idx_full_model),
+  baseline = shap_df_filtered$pred_card_shap_baseline_fmv[1]
+) %>%
+  shapviz::sv_importance(
+    "beeswarm",
+    max_display = 20L
+  ) +
   theme_minimal()
 ```
 
-## Thornton Beeswarm
+## Beeswarm Plot per Township
 
-```{r _shap_thornton_beeswarm, eval=shap_exists}
-shap_thornton_idx <- which(
-  assessment_card_filtered$meta_township_code == ccao::town_convert("Thornton")
-)
-create_shapviz(shap_df_filtered, assessment_card_filtered, shap_thornton_idx) %>%
-  shapviz::sv_importance("beeswarm") +
-  theme_minimal()
+::: {.panel-tabset}
+
+```{r _shap_beeswarm_iterate_township, results='asis', warning=F}
```

Comment:
The `warning=F` here is necessary because if you render this document by itself with warnings enabled, it messes up the tabset and makes it very difficult to understand how it will look in the final output.